torch/nested/_nestedtensor.py

```diff
@@ -0,0 +1,118 @@
+import torch
+import numbers
+
+
+def _filter_impl(args, kwargs):
+    if kwargs is None:
+        kwargs = {}
+    impl_args = []
+    for a in args:
+        if isinstance(a, torch.NestedTensor):
+            impl_args.append(a._impl)
+        elif torch.is_tensor(a):
+            impl_args.append(a)
+        elif isinstance(a, list):
+            a_impl, _ = _filter_impl(a, {})
+            impl_args.append(a_impl)
+        elif isinstance(a, tuple):
+            a_impl, _ = _filter_impl(a, {})
+            impl_args.append(tuple(a_impl))
+        else:
+            impl_args.append(a)
+    impl_kwargs = {
+        k: v._impl if isinstance(v, torch.NestedTensor) else v
+        for (k, v) in kwargs.items()
+    }
+    return impl_args, impl_kwargs
+
+
+def _wrap_result(result):
+    if isinstance(result, list):
+        return list(_wrap_result(r) for r in result)
+    if isinstance(result, tuple):
+        return tuple(_wrap_result(r) for r in result)
+    return (
+        torch.NestedTensor(result)
+        if torch.is_tensor(result) and torch.is_nt_impl(result)
+        else result
+    )
```

Comment:
Consider using tree_map here?

Reply:
I did, but remember it wasn't working. Will revisit.